File: server_integration_tests/clients/java_reactive_client/client-app/src/test/java/dev/graphlink/server/KeywordsTest.java
package dev.graphlink.server;

import dev.graphlink.server.generated.client.GraphLinkClient;
import dev.graphlink.server.generated.enums.Keyword;
import dev.graphlink.server.generated.inputs.ReservedInput;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

public class KeywordsTest {

    private GraphLinkClient client;

    @BeforeEach
    void setUp() {
        client = Fixtures.newClient();
    }

    @Test
    void reservedWordFieldsRoundTripBothJavaOnly() {
        var r = client.queries.reserved().block().getReserved();
        assertEquals("cls", r.getClass_());
        assertEquals(42, r.getReturn_());
        assertTrue(r.getNew_());
        assertEquals("def", r.getDefault_());
        assertEquals("yes", r.getIs());
        assertEquals("inside", r.getIn());
        assertEquals("w", r.getWith());
        assertEquals(7, r.getInt_());
        assertFalse(r.getSynchronized_());
        assertEquals("n", r.getNative_());
        assertEquals(Keyword.CLASS, r.getKind());
        assertEquals("v", r.getNested().getValue());
        assertEquals("tok", r.getSecret().getToken());
    }

    @Test
    void reservedOperationNamePlusReservedArgumentNames() {
        var res = client.queries.switch_("x", 5).block();
        assertEquals("x", res.getSwitch_().getClass_());
        assertEquals(5, res.getSwitch_().getReturn_());
    }

    @Test
    void leadingUnderscoreOperation() {
        var res = client.queries.status().block();
        assertEquals("ok", res.getStatus());
    }

    @Test
    void collidingFieldNamesStayDistinct() {
        var c = client.queries.collide().block().getCollide();
        assertTrue(List.of(c.getClass_2(), c.getClass2()).containsAll(List.of("A", "B")));
    }

    @Test
    void reservedWordInputFieldsMutation() {
        var res = client.mutations.echoReserved(
                ReservedInput.builder()
                        .class_("X").return_(5).default_("D").is(true).synchronized_(3).build()).block();
        assertEquals("X", res.getEchoReserved().getClass_());
        assertEquals(5, res.getEchoReserved().getReturn_());
        assertEquals("D", res.getEchoReserved().getDefault_());
    }
}

File: server_integration_tests/clients/kotlin_client/client-app/src/test/kotlin/dev/graphlink/server/KeywordsTest.kt
package dev.graphlink.server

import dev.graphlink.server.generated.client.GraphLinkClient
import dev.graphlink.server.generated.enums.Keyword
import dev.graphlink.server.generated.inputs.ReservedInput
import kotlinx.coroutines.test.runTest
import kotlin.test.BeforeTest
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class KeywordsTest {

    private lateinit var client: GraphLinkClient

    @BeforeTest
    fun setUp() {
        client = Fixtures.newClient()
    }

    @Test
    fun reservedWordFieldsRoundTripBothDartOnlyJavaOnly() = runTest {
        val r = client.queries.reserved().reserved
        assertEquals("cls", r.class_)
        assertEquals(42, r.return_)
        assertTrue(r.new)
        assertEquals("def", r.default)
        assertEquals("yes", r.is_)
        assertEquals("inside", r.in_)
        assertEquals("w", r.with)
        assertEquals(7, r.int)
        assertFalse(r.synchronized)
        assertEquals("n", r.native)
        assertEquals(Keyword.CLASS, r.kind)
        assertEquals("v", r.nested.value)
        assertEquals("tok", r.secret.token)
    }

    @Test
    fun reservedOperationNamePlusReservedArgumentNames() = runTest {
        val res = client.queries.switch(class_ = "x", return_ = 5)
        assertEquals("x", res.switch.class_)
        assertEquals(5, res.switch.return_)
    }

    @Test
    fun leadingUnderscoreOperation() = runTest {
        val res = client.queries.status()
        assertEquals("ok", res.status)
    }

    @Test
    fun collidingFieldNamesStayDistinct() = runTest {
        val c = client.queries.collide().collide
        assertTrue(listOf(c.class_2, c.class2).containsAll(listOf("A", "B")))
    }

    @Test
    fun reservedWordInputFieldsMutation() = runTest {
        val res = client.mutations.echoReserved(
            ReservedInput(class_ = "X", return_ = 5, default = "D", is_ = true, synchronized = 3),
        )
        assertEquals("X", res.echoReserved.class_)
        assertEquals(5, res.echoReserved.return_)
        assertEquals("D", res.echoReserved.default)
    }
}
